fastai/model.py

```diff
@@ -182,15 +188,14 @@ def hook(module, input, output):
            not isinstance(module, nn.ModuleList) and
            not (module == m)):
             hooks.append(module.register_forward_hook(hook))
-
     summary = OrderedDict()
     hooks = []
     m.apply(register_hook)
 
-    if isinstance(input_size[0], (list, tuple)):
-        x = [to_gpu(Variable(torch.rand(3,*in_size))) for in_size in input_size]
-    else: x = [to_gpu(Variable(torch.rand(3,*input_size)))]
-    m(*x)
+
+    if not isinstance(input_size[0], (list, tuple)): input_size = [input_size]
```

Comment:
what if input_size is simply an empty-list? Would this create problems? I'm not sure how to get around it though ..

Reply:
The input size have to be a  not empty list and it have to have more than 2 elements: (Number of Batches, size of a batch). Throwing OutOfRange in this example make sense as it will tell you that the function needs  a list that is not empty. If you want to have a more user friendly message we could add assert len(input_size) > 2, "The input_size have to have at least 2 elements"